src/mem_table.rs

```diff
@@ -12,11 +12,73 @@ use crate::bytes_range::BytesRange;
 use crate::error::SlateDBError;
 use crate::iter::{KeyValueIterator, SeekToKey};
 use crate::merge_iterator::MergeIterator;
-use crate::types::{RowAttributes, RowEntry, ValueDeletable};
+use crate::types::RowEntry;
 use crate::utils::WatchableOnceCell;
 
+/// Memtable may contains multiple versions of a single user key, with a monotonically increasing sequence number.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub(crate) struct KVTableInternalKey {
+    user_key: Bytes,
+    seq: u64,
+}
+
+impl KVTableInternalKey {
+    pub fn new(user_key: Bytes, seq: u64) -> Self {
+        Self { user_key, seq }
+    }
+}
+
+impl Ord for KVTableInternalKey {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.user_key
+            .cmp(&other.user_key)
+            .then(self.seq.cmp(&other.seq).reverse())
```

Comment:
Why reverse? Isn't `abc:123` < `abc:456`?

Reply:
on iteration & get, the entries with a higher sequence number are better to put in front, we can simply take the first entry to users.

`(user_key ASC, seq DESC)` is also the ordering of sst in leveldb (hopefully claude sonnet did not hallucinate 👀):

```
This sorting order ensures that for a given user key, the entry with the highest sequence number (which represents the most recent version of the key-value pair) comes first.

Here's an example to illustrate the sorting order:

Suppose we have the following (user key, sequence number) pairs:

("apple", 10)
("apple", 8)
("apple", 5)
("banana", 12)
("banana", 9)
("cherry", 7)
```